### ISSUE TITLE
RequiredPluginsClasspathContainer::getClasspathEntries doesn't consider indirect dependencies

### ISSUE THREAD

**Issue by sratz** [eclipse-pde/eclipse.pde#225]
This is a follow-up on https://bugs.eclipse.org/bugs/show_bug.cgi?id=534082 which is still a problem.

> 1)RequiredPluginsClasspathContainer::getClasspathEntries doesnt consider the indirect dependencies unless and until every bundle in the path to that dependency (including the end dependency) is re-exported.
> 2) Also it seems to not consider imported packages.

This causes issues when e.g. executing unit tests as plain JUnit test (without Eclipse platform running), where suddenly things are missing.

For more details see also https://bugs.eclipse.org/bugs/show_bug.cgi?id=537603

**Comment by laeubi** [comment]
This might also be related:
- https://github.com/eclipse-pde/eclipse.pde/issues/332

@sratz do you have a reproducer for this case:

> executing unit tests as plain JUnit test (without Eclipse platform running), where suddenly things are missing

**Comment by sratz** [comment]
@laeubi Yes, I have provided a reproducer in bug https://bugs.eclipse.org/bugs/show_bug.cgi?id=537603

**Comment by sratz** [comment]
There also was a proposed patch: https://git.eclipse.org/r/c/pde/eclipse.pde.ui/+/144502/
